IdentityCore/src/workplacejoin/MSIDRegistrationInformation.m

```diff
@@ -40,14 +41,18 @@ - (id)initWithSecurityIdentity:(SecIdentityRef)identity
                     privateKey:(SecKeyRef)privateKey
 
 {
-    if (!(identity && certificateIssuer && certificate && certificateSubject && certificateData && privateKey)) return nil;
+    if (!(certificateIssuer && certificate && certificateSubject && certificateData && privateKey)) return nil;
     
     if (!(self = [super init])) return nil;
     
     // ARC is not aware of Core Foundation objects, so they still have to be
     // manually retained and released.
-    _securityIdentity = identity;
-    CFRetain(identity);
+    if (identity)
```

Comment:
Why did we change this as it achieves the same behavior. The previous code first checks if identity is not nil, and then tries to retain it if it is non-nil. The same thing is done for certificate and private key.

Reply:
I checked the code again and I believe this is done because now this method can be called with or without identity. Can we do this in a different way as this method's name itself suggests initWithSecurityIdentity and having identity nil is not intuitive.:)

Reply:
refactored the code :)